go-controller/pkg/ovn/namespace.go

```diff
@@ -105,28 +105,48 @@ func (oc *Controller) multicastUpdateNamespace(ns *kapi.Namespace, nsInfo *names
 	}
 
 	var err error
+	nsInfo.multicastEnabled = enabled
 	if enabled {
 		err = oc.createMulticastAllowPolicy(ns.Name, nsInfo)
 	} else {
-		err = deleteMulticastAllowPolicy(ns.Name)
+		err = deleteMulticastAllowPolicy(ns.Name, nsInfo)
 	}
 	if err != nil {
 		klog.Errorf(err.Error())
 		return
 	}
-
-	nsInfo.multicastEnabled = enabled
 }
 
 // Cleans up the multicast policy for this namespace if multicast was
 // previously allowed.
 func (oc *Controller) multicastDeleteNamespace(ns *kapi.Namespace, nsInfo *namespaceInfo) {
 	if nsInfo.multicastEnabled {
-		if err := deleteMulticastAllowPolicy(ns.Name); err != nil {
+		nsInfo.multicastEnabled = false
+		if err := deleteMulticastAllowPolicy(ns.Name, nsInfo); err != nil {
 			klog.Errorf(err.Error())
 		}
 	}
-	nsInfo.multicastEnabled = false
+}
+
+// updateNamepacePortGroup updates the port_group applied to the namespace the port group is created when
+// required and only really deleted when all objects that use it are not present
+func (nsInfo *namespaceInfo) updateNamespacePortGroup(ns string) error {
+	if nsInfo.multicastEnabled {
+		// Multicast is enabled and the port group already created so there is nothing to do.
+		if nsInfo.portGroupUUID != "" {
+			return nil
```

Comment:
Nitpick: Don't write comments before `if` statements that assume the condition is true. You can't say "the port group [is] already created" on line 135 because you haven't checked that yet. Either move the comment to be inside the `if` body, or leave it where it is but make the comment itself conditional ("If the port group is already created there is nothing to do").

Reply:
done